### PR TITLE
hometask first

### PR DESCRIPTION
**Pull request by sofiia1998**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.6.1</version>
+        </dependency>
+        <dependency>
+            <groupId>repository.com.epam.tat.module4</groupId>
+            <artifactId>calculator</artifactId>
+            <version>1.0</version>
+            <scope>system</scope>
```
Comment (Dr-Java) [inline pom.xml:59]: It's not a good idea to use absolute paths in your project. Nobody will be able to run this. So, unfortunately,  it's impossible to check your HW now.
You'd better place this jar file somewhere in your project - in separate directory (e.g., resources) - and then provide a relative path.
So please, fix this issue first.

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <version>3.23.1</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
+        <plugins>
+<!--            <plugin>-->
```
Comment (Dr-Java) [inline pom.xml]: And why is it commented out? Checkstyle plugin is a must.

File: src/test/java/com/epam/tc/hw1/TestDiv.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        return new Object[][]{
+                {5.7, 2, 2.85},
+                {3.3, 3, 1.1},
+                {5.6, 0.8, 7}
+        };
+    }
+
+    @Test(dataProvider = "longData")
+    public void divTestLong(long first, long second, long third) {
+
```
Comment (Dr-Java) [inline src/test/java/com/epam/tc/hw1/TestDiv.java]: Please, move your calculator instance creation into a separate class. You could call it, e.g., CalculatorTestBase or whatever. And there you could create some set up (and tear down) method with annotation @BeforeClass (@AfterClass) from TestNG.
Then your concrete test classes could be inherited from this base class.

File: src/test/java/com/epam/tc/hw1/TestDiv.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    @DataProvider
+    public static Object[][] longData() {
+        return new Object[][]{
+                {20, 10, 2},
+                {5, 5, 1},
+                {100, 25, 4}
+        };
+    }
+
+    @DataProvider
```
Comment (Dr-Java) [inline src/test/java/com/epam/tc/hw1/TestDiv.java]: Please, move all your data providers into separate class or classes.

File: src/test/java/com/epam/tc/hw1/TestDiv.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.epam.tat.module4.Calculator;
+import org.assertj.core.data.Percentage;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
+public class TestDiv {
```
Comment (Dr-Java) [inline src/test/java/com/epam/tc/hw1/TestDiv.java]: Please, add case (data provider and test) with division by zero.

File: src/test/java/com/epam/tc/hw1/TestDiv.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
+package com.epam.tc.hw1;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.epam.tat.module4.Calculator;
+import org.assertj.core.data.Percentage;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+
```
Comment (Dr-Java) [inline src/test/java/com/epam/tc/hw1/TestDiv.java]: Let's name our test classes DivTest and so on, not TestDiv. Please, do the same for other classes.

**Review by Dr-Java — APPROVED** [review]
